### PR TITLE
feat(@spartacus/storefront): use generic page layout for reset password page

### LINKED ISSUE #758: [CAN'T REPRODUCE] Form stops displaying when creating shipping address
From #730 - smoke testing Dec 18 lib release
After creating a Shipping Address, (moving on to step 2: Shipping Method), I was not able to continue. When I wen't back to Step 1: Shipping Address, nothing appears and now I am stuck

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
closes #758, #1248

### CODE REVIEW

File: projects/storefrontlib/src/lib/cms/page-layout/page-layout.service.ts [hunk 1]
```diff
@@ -63,7 +69,10 @@ export class PageLayoutService {
   }
 
   get page$(): Observable<Page> {
-    return this.cms.getCurrentPage().pipe(filter(Boolean));
+    return this.cms.getCurrentPage().pipe(
+      filter(Boolean),
+      tap(p => console.log(p))
```
Comment (Platonn) [inline projects/storefrontlib/src/lib/cms/page-layout/page-layout.service.ts]: Was this `console.log` intended?

File: projects/storefrontlib/src/lib/cms-lib/address-book/address-book.component.html [hunk 2]
```diff
@@ -1,67 +1,50 @@
...
+  </div>
+</div>
+
+<div class="cx-section" *ngIf="checkIfAnyFormOpen()">
+  <section #addAddressForm *ngIf="isAddAddressFormOpen">
+    <p>Add a new shipping address</p>
+    <cx-address-form
+      actionBtnLabel="Add address"
+      cancelBtnLabel="Back to address list"
+      [setAsDefaultField]="!(!addresses || addresses.length === 0)"
```
Comment (Platonn) [inline projects/storefrontlib/src/lib/cms-lib/address-book/address-book.component.html]: It could be simplified  to: `addresses && addresses.length > 0`

Comment (Platonn) [reply projects/storefrontlib/src/lib/cms-lib/address-book/address-book.component.html]: I cannot find any reference to simply `addresses`, we have only the stream `addresses$` in this scope.

File: projects/storefrontstyles/scss/layout/_index.scss [hunk 3]
```diff
@@ -33,3 +33,8 @@ cx-page-layout {
 .LoginPageTemplate {
   @extend %LoginPageTemplate !optional;
 }
+
+.AccountPageTemplate {
+  --cx-max-width: 100%;
+  @extend %LoginPageTemplate !optional;
```
Comment (Platonn) [inline projects/storefrontstyles/scss/layout/_index.scss]: I'm wondering if the LoginPageTemplate should extend the AccountPageTemplate?

File: projects/storefrontlib/src/lib/cms-lib/address-book/address-book.component.html [hunk 4]
```diff
@@ -1,67 +1,50 @@
-<div class="cx-page">
-  <section class="cx-page__section container">
-    <div class="row" *ngIf="(addresses$ | async) as addresses">
```
Comment (Platonn) [inline projects/storefrontlib/src/lib/cms-lib/address-book/address-book.component.html]: Was this `*ngIf` removed on purpose? It probably prevented the DOM sub-tree from being rendered before data are loaded.

File: projects/storefrontstyles/scss/layout/_index.scss (hunk 3 above)
Comment (Platonn) [reply projects/storefrontstyles/scss/layout/_index.scss]: Or why `.AccountPageTemplate` class doesn't extend it's own `%AccountPageTemplate`?

Comment (tobi-or-not-tobi) [reply projects/storefrontstyles/scss/layout/_index.scss]: Yes, its a bit in between situation. Once all page templates are moved to this approach, i'm planning on eliminating backend templates. The front-end will probably lead, and AccountPageTemplate, LoginPageTemplate, NotFound, etc. are going to be merged. I need to come with a good name, and then it's time to move. I'm delaying this decision, first want to have good visibility on all of them.

File: projects/storefrontlib/src/lib/cms-lib/address-book/address-book.component.html (hunk 4 above)
Comment (tobi-or-not-tobi) [reply projects/storefrontlib/src/lib/cms-lib/address-book/address-book.component.html]: Ah, there it is. The missing piece I was looking for. I accidentally removed it, and then found an issue that I thought I fixed. I'll have a 2nd look and commit a change.

**Pull request merged**